Blockzilla/SettingsViewController.swift

```diff
@@ -348,22 +350,27 @@ class SettingsViewController: UIViewController, UITableViewDataSource, UITableVi
             let identifier = indexPath.row == 0 ? "SettingsViewController.searchCell" : "SettingsViewController.autocompleteCell"
 
             searchCell.accessoryLabelText = accessoryLabel
-            searchCell.label = label
+            searchCell.labelText = label
             searchCell.accessibilityIdentifier = identifier
 
             cell = searchCell
         case .siri:
             guard let siriCell = tableView.dequeueReusableCell(withIdentifier: "accessoryCell") as? SettingsTableViewAccessoryCell else { fatalError("No Search Cells!") }
             if indexPath.row == 0 {
-                siriCell.label = UIConstants.strings.eraseSiri
-                siriCell.accessoryLabelText = UIConstants.strings.addToSiri
+                siriCell.labelText = UIConstants.strings.eraseSiri
                 siriCell.accessibilityIdentifier = "settingsViewController.siriEraseCell"
             } else if indexPath.row == 1 {
-                siriCell.label = String(format: UIConstants.strings.eraseAndOpenSiri, AppInfo.productName)
-                siriCell.accessoryLabelText = UIConstants.strings.addToSiri
-                siriCell.accessibilityIdentifier = "settingsViewController.siriEraseAndOpenCell"
+                if #available(iOS 12.0, *) {
+                    siriCell.labelText = UIConstants.strings.eraseAndOpenSiri
+                    siriCell.accessibilityIdentifier = "settingsViewController.siriEraseAndOpenCell"
+                    hasAddedActivity(type: .eraseAndOpen) { (result: Bool) in
+                        DispatchQueue.main.async {
```

Comment:
It is better to reduce the surface area multi-threaded code.`hasAddActivity` has no indication the completion would be off main, ensure all completion blocks happen on-main by default (unless documented otherwise, typically for performance reasons or due to blocking APIs under the hood.)

Reply:
Fair point - thanks for explaining. Will resolve.